ui/app/components/console/ui-panel.js

```diff
@@ -81,17 +87,20 @@ export default Component.extend({
     const inputError = formattedErrorFromInput(path, method, flags, dataArray);
     if (inputError) {
       this.logAndOutput(command, inputError);
+      waiter.endAsync(waiterToken);
       return;
     }
     try {
       const resp = yield service[camelize(method)].call(service, path, data, flags);
       this.logAndOutput(command, logFromResponse(resp, path, method, flags));
     } catch (error) {
       if (error instanceof ControlGroupError) {
+        waiter.endAsync(waiterToken);
         return this.logAndOutput(command, this.controlGroup.logFromError(error));
```

Comment:
It looks like all of the paths in `executeCommand` end with calling `logAndOutput`. I wonder if it would be better to pass the `waitToken` into that method and do `endAsync` there instead. I'm just wondering if there could be some side effects here where the test continues by `logAndOutput` is still doing things.

Reply:
Good idea. Let me give that a go.

Reply:
I gave it a go and hit test errors. Going to keep as is.

Reply:
Sounds good. That whole chain is a bit complex so that's not too surprising.